File: sandbox-runner/patterns/event-accumulator/go/starter.go
package main

import (
	"context"
	"fmt"
	"log"

	"go.temporal.io/sdk/client"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	// Items arriving for two separate orders from multiple producers.
	// item-1 is sent twice to demonstrate deduplication.
	items := []OrderItem{
		{OrderID: "order-A", ItemID: "item-1", Name: "Widget"},
		{OrderID: "order-B", ItemID: "item-2", Name: "Gadget"},
		{OrderID: "order-A", ItemID: "item-3", Name: "Gizmo"},
		{OrderID: "order-B", ItemID: "item-4", Name: "Doohickey"},
		{OrderID: "order-A", ItemID: "item-1", Name: "Widget"}, // duplicate — ignored
		{OrderID: "order-A", ItemID: "item-5", Name: "Thingamajig"},
	}

	for _, item := range items {
		workflowID := fmt.Sprintf("%s-%s", WorkflowIDPrefix, item.OrderID)
		// Signal-With-Start: start the workflow if not running, otherwise just signal
		_, err := c.SignalWithStartWorkflow(
			context.Background(),
			workflowID,
			"add-item",
			item,
			client.StartWorkflowOptions{
				ID:        workflowID,
				TaskQueue: TaskQueue,
			},
			AccumulatorWorkflow,
			item.OrderID,
			[]OrderItem{},
			[]string{},
		)
		if err != nil {
			log.Fatalf("SignalWithStartWorkflow failed: %v", err)
		}
		fmt.Printf("Signaled %s: %s (%s)\n", workflowID, item.Name, item.ItemID)
	}

	fmt.Println("\nWaiting for workflows to process their items...")
	for _, orderID := range []string{"order-A", "order-B"} {
		workflowID := fmt.Sprintf("%s-%s", WorkflowIDPrefix, orderID)
		run := c.GetWorkflow(context.Background(), workflowID, "")
		var result string
		if err := run.Get(context.Background(), &result); err != nil {
			log.Fatalf("Workflow %s failed: %v", workflowID, err)
		}
		fmt.Printf("%s result: %s\n", workflowID, result)
	}
	fmt.Printf("\nOpen the Temporal UI and search for '%s' to see the accumulator workflows.\n", WorkflowIDPrefix)
}

File: sandbox-runner/patterns/event-accumulator/java/src/main/java/Starter.java
import io.temporal.client.BatchRequest;
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;

import java.util.ArrayList;
import java.util.List;

public class Starter {
    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        // Items arriving for two separate orders from multiple producers.
        // item-1 is sent twice to demonstrate deduplication.
        List<Shared.OrderItem> items = List.of(
                new Shared.OrderItem("order-A", "item-1", "Widget"),
                new Shared.OrderItem("order-B", "item-2", "Gadget"),
                new Shared.OrderItem("order-A", "item-3", "Gizmo"),
                new Shared.OrderItem("order-B", "item-4", "Doohickey"),
                new Shared.OrderItem("order-A", "item-1", "Widget"),   // duplicate — ignored
                new Shared.OrderItem("order-A", "item-5", "Thingamajig")
        );

        for (Shared.OrderItem item : items) {
            String workflowId = Shared.WORKFLOW_ID_PREFIX + "-" + item.orderId;
            AccumulatorWorkflow stub = client.newWorkflowStub(
                    AccumulatorWorkflow.class,
                    WorkflowOptions.newBuilder()
                            .setTaskQueue(Shared.TASK_QUEUE)
                            .setWorkflowId(workflowId)
                            .build());

            // Signal-With-Start: start if not running, otherwise just deliver the signal
            BatchRequest request = client.newSignalWithStartRequest();
            request.add(stub::accumulate, item.orderId, new ArrayList<>(), new ArrayList<>());
            request.add(stub::addItem, item);
            client.signalWithStart(request);

            System.out.println("Signaled " + workflowId + ": " + item.name + " (" + item.itemId + ")");
        }

        System.out.println("\nWaiting for workflows to process their items...");
        for (String orderId : new String[]{"order-A", "order-B"}) {
            String workflowId = Shared.WORKFLOW_ID_PREFIX + "-" + orderId;
            // Connect to the running workflow by ID and wait for its result
            AccumulatorWorkflow handle = client.newWorkflowStub(AccumulatorWorkflow.class, workflowId);
            String result = handle.accumulate(orderId, new ArrayList<>(), new ArrayList<>());
            System.out.println(workflowId + " result: " + result);
        }
        System.out.println("\nOpen the Temporal UI and search for '" + Shared.WORKFLOW_ID_PREFIX
                + "' to see the accumulator workflows.");
    }
}
